### ISSUE TITLE
Training error

### ISSUE THREAD

**Issue by ding-s-h** [Johswald/Bayesian-FlowNet#1]
Hi @Johswald thanks for sharing the work.
I tried the code but had difficulty in convergence. What is your training error and how long (iterations) does it take for you to converge? Thanks.

**Comment by Johswald** [comment]
thanks for your interest! sorry for my late reply.
Just uploaded my current state, this is a work in progress.
At least 600t steps (batch size 8) are necessary but I still have to fine tune (check todos in architecure) and test/simplify my data augmentation.

I trained for couple 100t steps without data augmentation and already get good results on the test set. Please get back to me if you find nice hyperparameters or any mistakes.
(Somehow I didn't figure out yet to simply calc the aae error in eval.py...) Thanks!

**Comment by ding-s-h** [comment]
Thanks for the reply! It seems like I didn't train long enough, not nearly close to 600t steps.
I did change the Relu to leaky Relu with leak=0.1 to match the Caffe version though, using the implementation in [this issue](https://github.com/tensorflow/tensorflow/issues/4079)
I also have to change the line in [readFlowFile.py](https://github.com/Johswald/FlowNet_Tensorflow/blob/master/readFlowFile.py)
`data = np.fromfile(f, np.float32, count=2*w*h)`
to 
`data = np.fromfile(f, np.float32, count=2*w[0]*h[0])`
to make it work.
BTW, what is the output resolution of "flow_predict" in architectures.py before resizing/upsampling, is it 96x128 or 192x256? Thank you.

**Comment by Johswald** [comment]
thanks for the leaky Relu hint, will try that on/off as well. 
Check flownet_tools.py, I used the tf.FixedLengthRecordReader to read .flo files directly. 
Should be faster.
Last output is 96x128, check the paper for a more sophisticated upsample.

**Comment by Johswald** [comment]
ah, readFlowFile.py works fine on my side. a numpy version issue?

**Comment by ding-s-h** [comment]
I also used the tf.FixedLengthRecordReader to read .flo files, was just using readFlowFile.py for some separate visualization. I'm using python3 if that matters.
For me, the network finally converges after I rescale image value to [-0.5, 0.5], and multiply the flow value by 0.05, to make it roughly in [-0.5, 0.5] too. Took me 100k~200k iterations with a batch size of 16. Consider this issue closed?

**Comment by Johswald** [comment]
could I see your final architecture? johannes.oswald@tum.de
sure, thanks!

**Comment by ding-s-h** [comment]
@Johswald sure, just uploaded my architecture [here](https://github.com/DingGit/flownet-tf/blob/master/flownet.py).
